lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -635,6 +635,46 @@ class ConvertAtenSqueezeAllDimsOp : public ConvertAtenSqueezeOp<AtenOpT> {
   }
 };
 
+// FIXME(AG): This will eventually go into a Tosa*Utils file
+// Convert an fp32 scalar into tosa fp32 tensor.
+static LogicalResult
+tosaF32TensorFromTorchFloat(ConversionPatternRewriter &rewriter, Operation *op,
+                            Value torchScalarValue, Value &tosaTensor) {
+  double scalarValue;
+
+  if (!matchPattern(torchScalarValue, m_TorchConstantFloat(&scalarValue)))
+    return failure();
+
+  // Construct a tosa.const
+  tosaTensor =
+      mlir::tosa::getTosaConstTensorSingleF32(rewriter, op, scalarValue);
+
+  return success();
+}
+
+template <>
+LogicalResult ConvertAtenOp<AtenPowTensorScalarOp>::matchAndRewrite(
+    AtenPowTensorScalarOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  Value self = adaptor.self();
+  auto selfTy = self.getType().template cast<RankedTensorType>();
```

Comment:
I think we missed a type check `getElementType().isa<mlir::FloatType>()` here. sry, I missed this in the previous review.

Reply:
'tosaF32TensorFromTorchFloat' will fail if the 'exponent' does not parse as a float.

Reply:
We'd prefer an assert to make it explicit only floating point is supported. Failure in `tosaF32TensorFromTorchFloat` is a symptom of passing in wrong input type but doesn't really indicate the root cause with out some level of investigation.